components/src/core/DropdownToolbar/DropdownToolbar.tsx

```diff
@@ -0,0 +1,190 @@
+    subtitleContent?: string;
+    subtitle?: string;
+    arrowDropdown?: string;
+    menuItem?: string;
+};
+
+export const DropdownToolbar: React.FC<DropdownToolbarProps> = (props) => {
+    const { title, subtitle, menuItems, navigationIcon, customMenu, classes = {}, ...toolbarProps } = props;
+    const [anchorEl, setAnchorEl] = useState(null);
+    const anchor = useRef(null);
+    const theme = useTheme();
+    const defaultClasses = useStyles(theme);
+    const getNavigationIcon = useCallback(() => {
+        if (navigationIcon) {
+            return <div className={clsx(defaultClasses.navigation, classes.navigation)}>{navigationIcon}</div>;
+        }
+    }, [navigationIcon]);
+
+    const getMenu = useCallback(() => {
+        if (menuItems && Boolean(anchorEl)) {
+            return (
+                <Menu
+                    elevation={0}
+                    getContentAnchorEl={null}
+                    anchorOrigin={{ vertical: 'bottom', horizontal: 'left' }}
+                    transformOrigin={{ vertical: 'top', horizontal: 'left' }}
+                    anchorEl={anchorEl}
+                    open={Boolean(anchorEl)}
+                    onClose={(): void => setAnchorEl(null)}
+                    MenuListProps={{ style: { padding: 0 } }}
+                    PaperProps={{
+                        style: {
+                            maxHeight: MENU_ITEM_HEIGHT * 4.5,
+                        },
+                    }}
+                >
+                    {!customMenu &&
+                        menuItems.map((item: ToolbarMenuItem, index: number) => (
+                            <MenuItem
+                                key={`Toolbar-Option-${index}`}
+                                onClick={(): void => {
+                                    setAnchorEl(null);
+                                    item.onClick();
+                                }}
+                                className={clsx(classes.menuItem)}
+                            >
+                                {item.label}
+                            </MenuItem>
+                        ))}
+                </Menu>
+            );
+        }
+
+        if (customMenu && Boolean(anchorEl)) {
+            return (
+                <Menu
+                    elevation={0}
+                    getContentAnchorEl={null}
+                    anchorOrigin={{ vertical: 'bottom', horizontal: 'left' }}
+                    transformOrigin={{ vertical: 'top', horizontal: 'left' }}
+                    anchorEl={anchorEl}
+                    open={Boolean(anchorEl)}
+                    onClose={(): void => setAnchorEl(null)}
+                    MenuListProps={{ style: { padding: 0 } }}
+                    PaperProps={{
+                        style: {
+                            maxHeight: MENU_ITEM_HEIGHT * 4.5,
+                        },
+                    }}
+                >
+                    {customMenu}
+                </Menu>
```

Comment:
If they are specifying a custom menu, they will provide the entire `<Menu>` component rather than just the contents of it. Take a look at how the UserMenu does this...we clone the menu that they pass in and then add the open/close/anchor logic that we need to show it.